applepay/utils.py

```diff
@@ -196,70 +202,90 @@ def get_provided_data(token):
+    Returns:
+        string: The BER-encoded signed attributes as a string
+            of bytes
+    """
     class_ = super(signed_attrs.__class__, signed_attrs).class_
     method = super(signed_attrs.__class__, signed_attrs).method
     tag = super(signed_attrs.__class__, signed_attrs).tag
 
     return emit(class_, method, tag, signed_attrs.contents)
 
 
-def get_public_key_bytes(leaf_cert):
-    """TODO
+def remove_ec_point_prefix(point):
+    """Remove the prefix from the uncompressed ec point.
 
-    We expect the public key to be in the uncompressed
+    We expect the point to be in the uncompressed
     format described here: https://tools.ietf.org/html/rfc5480#section-2.2
     This means the first byte must be "\x04" otherwise consider
-    the public key not usable.public_key_bytes
+    the public key not usable.  ecdsa.keys.VerifyingKey does not
+    handle this first byte so its chopped off before turning the public
+    key bytes.
 
-    Return the public key value minus the byte prefix for use with
-    the ecdsa.keys.VerifyingKey class
+    Args:
+        point (string): The uncompressed byte string of an EC Point
+    Returns:
+        string: The EC point byte string minus the uncompressed
+            byte prefix indicator
     """
-    public_key_bytes = leaf_cert.public_key['public_key'].native
-
-    if not public_key_bytes.startswith("\x04"):
-        logger.warning("Expected uncompressed public key bytes.")
+    if not point.startswith("\x04"):
+        logger.warning("Expected uncompressed EC point.")
         return None
 
-    return public_key_bytes[1:]
+    return point[1:]
 
 
 def get_first_from_iterable(filter_func, iterable):
-    """TODO"""
+    """Get the first filtered item from an iterable.
+
+    Args:
+        filter_func (callable): the function to filter on
+        iterable (iterable): the iterable to filter on
+    Returns:
+        object: the first filtered item from iterable or None
+            if no items matching the filter are found
+    """
     filtered_stuff = ifilter(filter_func, iterable)
     return next(filtered_stuff, None)
 
 
-def get_callable_hashfunc_by_name(name, data):
-    """TODO
-    Return a callable hashfunc by name
-    using the hashlib.new constructor
+def get_hashfunc_by_name(name, data):
+    """
+    Get a callable hashfunc by name.
+
+    This function can be used directly or with functools.partial, for example:
+
+    >>> hashfunc = functools.partial(get_hashfunc_by_name, 'sha256')
+    >>> hashfunc('sir robin').digest()
+
+    Args:
+        name (string): The string name of the desired algorithm
+        data (buffer): The buffer to hash
+    Returns:
+        callable: The callable hash function of the provided
+            algorithm updated with the data to be hashed
     """
     hashfunc = hashlib.new(name)
     hashfunc.update(data)
     return hashfunc
 
 
-def signature_is_valid(token):
-    """TODO write me
+def verify_signature(token):
```

Comment:
I think we should be using function `signing_time_is_valid` from line 56 somewhere inside of `verify_signature` to meet the requirement from apple docs 
`Inspect the CMS signing time of the signature, as defined by section 11.3 of RFC 5652. If the time signature and the transaction time differ by more than a few minutes, it's possible that the token is a replay attack.`